src/file/InputStream.cpp

```diff
@@ -69,12 +69,17 @@ void InputStream::read(char *buffer, int64_t length) {
             needUpdate = true;
         }
 
+        if(read <= 0){
```

Comment:
We should check the actual remaining length in File level. In this place, we only read length that File asked.